src/popup.js

```diff
@@ -519,6 +519,35 @@ window.addEventListener('load', function(evt) {
 
     window.close();
   });
+  
+  $(".wcmmode").click(function(e) {
+    e.preventDefault();
+    
+    var $this = $(this),
+        url = getUrlWithUpdatedQueryString(pageDetails.location, null),
+        wcmMode = $this.attr("data-wcmmode");
+    
+    var newCookie = {
+        url: url,
+        name: "wcmmode",
+        value: wcmMode,
+        path: "/",
+        //domain: domain,
+        secure: url.slice(0,5) === "https"
+    };
+    chrome.cookies.getAll({}, function(cookies) {
+        var wcmModeCookie = cookies.filter(function(cookie) {
+            return cookie.name === "wcmmode";
+        });
+        
+        if (!wcmModeCookie.length || wcmModeCookie[0].value !== wcmMode) {
+            newCookie.value = wcmMode
+            chrome.cookies.set(newCookie, function() {
+                chrome.tabs.reload();
```

Comment:
I think that if we change over to cookies from querystring params, we should take the responsibility of checking the URL for the "wcmmode" parameter and either 1) updating it to match or 2) remove it completely. 

Even with the plugin, I'll still type in the address bar - I think it depends on where my hands are. It will be confusing if the cookie forces AEM to behave one way and the visible address bar says something different.

To remove the param and update the browser: 
setTabLocation(getUrlWithUpdatedQueryString(pageDetails.location, 'wcmmode'))

To change the param and update the browser:
setTabLocation(getUrlWithUpdatedQueryString(pageDetails.location, 'wcmmode', 'edit'))

We may need to update getUrlWithUpdatedQueryString to deal with upper/lower case.

Thank you!